packages/create-cloudflare/templates-experimental/astro/c3.ts

```diff
@@ -59,43 +58,28 @@ const updateAstroConfig = () => {
 	});
 };
 
-const updateEnvDeclaration = (ctx: C3Context) => {
-	if (!usesTypescript(ctx)) {
-		return;
-	}
-
-	const filePath = "src/env.d.ts";
-
-	updateStatus(`Adding type declarations in ${blue(filePath)}`);
-
-	transformFile(filePath, {
-		visitProgram: function (n) {
-			const snippets = loadTemplateSnippets(ctx);
-			const patch = snippets.runtimeDeclarationTs;
-			const b = recast.types.builders;
-
-			// Preserve comments with the new body
-			const comments = n.get("comments").value;
-			n.node.comments = comments.map((c: recast.types.namedTypes.CommentLine) =>
-				b.commentLine(c.value),
-			);
-
-			// Add the patch
-			n.get("body").push(...patch);
-
-			return false;
-		},
-	});
-};
-
 const config: TemplateConfig = {
 	configVersion: 1,
 	id: "astro",
 	frameworkCli: "create-astro",
 	platform: "workers",
 	displayName: "Astro",
 	copyFiles: {
-		path: "./templates",
+		async selectVariant(ctx) {
+			// Note: this `selectVariant` function should not be needed
+			//       this is just a quick workaround until
+			//       https://github.com/cloudflare/workers-sdk/issues/7495
+			//       is resolved
+			return usesTypescript(ctx) ? "ts" : "js";
+		},
```

Comment:
Funnily enough this isn't really needed since it looks like the Astro CLI creates a tsconfig.json for you even if you tell it that you're not planning on using TypeScript 🤷

I would still keep this in as I think this is the proper implementation (workaround excluded) and it is more robust (it won't break if the Astro CLI changes the above behavior and stops creating the tsconfig file)